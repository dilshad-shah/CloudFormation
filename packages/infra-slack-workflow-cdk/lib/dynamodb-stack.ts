import { Construct, RemovalPolicy, Stack, StackProps } from '@aws-cdk/core';
import * as dynamodb from '@aws-cdk/aws-dynamodb';
import { IFunction } from '@aws-cdk/aws-lambda';
import { addLintIgnore } from '@cvent/aws-cdk-core';

export interface InfraSlackWorkflowDynamoStackProps extends StackProps {
  tableName: string;
  partitionKey: string;
  infraSlackWorkflowLambda: IFunction;
  cventEnvironment: string;
}

export class InfraSlackWorkflowDynamoStack extends Stack {
  constructor(scope: Construct, id: string, props?: InfraSlackWorkflowDynamoStackProps) {
    super(scope, id, props);

    const table = new dynamodb.Table(this, 'infra-slack-workflow-dynamodb', {
      partitionKey: { name: 'threadId', type: dynamodb.AttributeType.STRING },
      tableName: props.tableName,
      removalPolicy: RemovalPolicy.DESTROY
    });

    addLintIgnore(table, ['E9020', 'E9019', 'E9022']);

    table.grantReadWriteData(props.infraSlackWorkflowLambda);
  }
}
