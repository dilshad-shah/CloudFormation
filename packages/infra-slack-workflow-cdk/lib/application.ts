import { InfraSlackWorkflowStack } from './stack';
import { InfraSlackWorkflowDynamoStack } from './dynamodb-stack';
import { ephemeralize, generateTags, stackName } from '@cvent/aws-cdk-core';
import { Env, SubEnv } from '@cvent/environments';
import { App, Environment } from '@aws-cdk/core';

export interface ApplicationProps {
  /** used primarily for CI stacks to give them a unique name based on package version */
  readonly version?: string;

  /** AWS account and region */
  readonly awsEnvironment: Environment;

  /** name of the hogan environment (e.g. pr50) */
  readonly cventEnvironment: string;

  /** name of the Cvent Sub-environment (e.g. production) */
  readonly cventSubEnv?: SubEnv;

  readonly tableName?: string;
  readonly partitionKey?: string;
  // readonly infraWorkflowLambda?: IFunction;

  readonly lambdaEnvJira?: string;
  readonly lambdaEnvPd?: string;
  readonly lambdaEnvSlack?: string;
  readonly lambdaEnvDBTable?: string;
  readonly lambdaEnvPdSrvId?: string;
}

export class Application extends App {
  // Exposed for testing purposes
  public readonly stack: InfraSlackWorkflowStack;
  public readonly dynamodbStack: InfraSlackWorkflowDynamoStack;

  constructor(id: string, props: ApplicationProps) {
    super();

    this.stack = new InfraSlackWorkflowStack(this, stackName(id, props.version), {
      env: props.awsEnvironment,
      lambdaEnvJira: props.lambdaEnvJira,
      lambdaEnvPd: props.lambdaEnvPd,
      lambdaEnvSlack: props.lambdaEnvSlack,
      lambdaEnvDBTable: props.lambdaEnvDBTable,
      lambdaEnvPdSrvId: props.lambdaEnvPdSrvId
    });

    generateTags(this.stack, {
      product: 'shared',
      businessUnit: 'shared',
      platform: 'shared',
      env: props.cventEnvironment as Env,
      subEnv: props.cventSubEnv
    });

    if (props.version !== undefined) {
      this.dynamodbStack = new InfraSlackWorkflowDynamoStack(this, stackName(id + '-dynamodb-' + props.version), {
        env: props.awsEnvironment,
        tableName: props.tableName,
        partitionKey: props.partitionKey,
        infraSlackWorkflowLambda: this.stack.infraSlackWorkflowLambda,
        cventEnvironment: props.cventEnvironment
      });
    } else {
      this.dynamodbStack = new InfraSlackWorkflowDynamoStack(this, stackName(id + '-dynamodb'), {
        env: props.awsEnvironment,
        tableName: props.tableName,
        partitionKey: props.partitionKey,
        infraSlackWorkflowLambda: this.stack.infraSlackWorkflowLambda,
        cventEnvironment: props.cventEnvironment
      });
    }

    generateTags(this.dynamodbStack, {
      product: 'shared',
      businessUnit: 'shared',
      platform: 'shared',
      env: props.cventEnvironment as Env,
      subEnv: props.cventSubEnv
    });

    if (props.cventEnvironment === 'ci') {
      ephemeralize(this.stack);
      ephemeralize(this.dynamodbStack);
    }
  }
}
