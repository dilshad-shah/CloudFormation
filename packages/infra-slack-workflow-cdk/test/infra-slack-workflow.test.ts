import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import { Application } from '../lib/application';

describe('application', (): void => {
  test('with defaults', (): void => {
    const app = new Application('infra-slack-workflow', {
      awsEnvironment: { account: '572724207364', region: 'us-east-1' },
      cventEnvironment: 'beach',
      cventSubEnv: 'beach'
    });

    expect(SynthUtils.toCloudFormation(app.stack)).toMatchSnapshot();
  });

  test('with version', (): void => {
    const app = new Application('infra-slack-workflow', {
      awsEnvironment: { account: '572724207364', region: 'us-east-1' },
      cventEnvironment: 'beach',
      cventSubEnv: 'beach',
      version: 'version'
    });

    expect(SynthUtils.toCloudFormation(app.stack)).toMatchSnapshot();
  });
});
