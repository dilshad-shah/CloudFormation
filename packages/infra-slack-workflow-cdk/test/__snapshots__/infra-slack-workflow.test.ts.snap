// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application with defaults 1`] = `
Object {
  "Outputs": Object {
    "LoadBalancerDNS": Object {
      "Value": Object {
        "Ref": "infraslackworkflowapiDomainDnsRecordCFDC8FA1",
      },
    },
    "infraslackworkflowapiEndpoint05FBE7B7": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "infraslackworkflowapiB910C56A",
            },
            ".execute-api.us-east-1.",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "infraslackworkflowapiDeploymentStageprodD67E1A6B",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecArtifactHashF1D0A81E": Object {
      "Description": "Artifact hash for asset \\"933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ec\\"",
      "Type": "String",
    },
    "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3BucketB74DFC99": Object {
      "Description": "S3 bucket for asset \\"933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ec\\"",
      "Type": "String",
    },
    "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3VersionKeyCAADF052": Object {
      "Description": "S3 key for asset version \\"933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ec\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "InfraSlackWorkflowLambda833DA3E4": Object {
      "DependsOn": Array [
        "InfraSlackWorkflowLambdaServiceRoleDefaultPolicy94F58139",
        "InfraSlackWorkflowLambdaServiceRole75DBBB72",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3BucketB74DFC99",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3VersionKeyCAADF052",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3VersionKeyCAADF052",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {},
        },
        "Handler": "index.handler",
        "MemorySize": 3008,
        "Role": Object {
          "Fn::GetAtt": Array [
            "InfraSlackWorkflowLambdaServiceRole75DBBB72",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "InfraSlackWorkflowLambdaEventInvokeConfigD2210958": Object {
      "Properties": Object {
        "FunctionName": Object {
          "Ref": "InfraSlackWorkflowLambda833DA3E4",
        },
        "MaximumRetryAttempts": 0,
        "Qualifier": "$LATEST",
      },
      "Type": "AWS::Lambda::EventInvokeConfig",
    },
    "InfraSlackWorkflowLambdaServiceRole75DBBB72": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "InfraSlackWorkflowLambdaServiceRoleDefaultPolicy94F58139": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ssm:GetParameter",
                "ssm:GetParameters",
              ],
              "Effect": "Allow",
              "Resource": Array [
                "arn:aws:ssm:us-east-1:572724207364:parameterundefined",
              ],
            },
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::ImportValue": "infra-slack-workflow-dynamodb:ExportsOutputFnGetAttinfraslackworkflowdynamodbCB872367Arn36B4C1E3",
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "InfraSlackWorkflowLambdaServiceRoleDefaultPolicy94F58139",
        "Roles": Array [
          Object {
            "Ref": "InfraSlackWorkflowLambdaServiceRole75DBBB72",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "infraslackworkflowapiB910C56A": Object {
      "Properties": Object {
        "Description": "API for the InfraWorkflow App",
        "EndpointConfiguration": Object {
          "Types": Array [
            "REGIONAL",
          ],
        },
        "Name": "infra-slack-workflow-api",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "infraslackworkflowapiDeployment97D3C2AFa0112814ea588aefd026eba378ffe5f2": Object {
      "DependsOn": Array [
        "infraslackworkflowapieventhandlerPOSTA78C202D",
        "infraslackworkflowapieventhandler2B3DE92B",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "infraslackworkflowapiDeploymentStageprodD67E1A6B": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "infraslackworkflowapiDeployment97D3C2AFa0112814ea588aefd026eba378ffe5f2",
        },
        "MethodSettings": Array [
          Object {
            "DataTraceEnabled": false,
            "HttpMethod": "*",
            "LoggingLevel": "INFO",
            "MetricsEnabled": true,
            "ResourcePath": "/*",
          },
        ],
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
        "StageName": "prod",
        "TracingEnabled": true,
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "infraslackworkflowapiDomain1F140300": Object {
      "Properties": Object {
        "DomainName": "infraslackworkflowstack.cvent-sandbox.cvent.cloud",
        "EndpointConfiguration": Object {
          "Types": Array [
            "REGIONAL",
          ],
        },
        "RegionalCertificateArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":acm:us-east-1:572724207364:certificate/56e71c9e-f5a0-47d7-9c10-a11c0fbbf502",
            ],
          ],
        },
      },
      "Type": "AWS::ApiGateway::DomainName",
    },
    "infraslackworkflowapiDomainDnsRecordCFDC8FA1": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "infraslackworkflowapiDomain1F140300",
              "RegionalDomainName",
            ],
          },
          "HostedZoneId": Object {
            "Fn::GetAtt": Array [
              "infraslackworkflowapiDomain1F140300",
              "RegionalHostedZoneId",
            ],
          },
        },
        "HostedZoneId": "Z321UXK1TZE4RD",
        "Name": "infraslackworkflowstack.cvent-sandbox.cvent.cloud.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "infraslackworkflowapiDomainMapinfraslackworkflowinfraslackworkflowapi7A68E0753D68EF10": Object {
      "Properties": Object {
        "DomainName": Object {
          "Ref": "infraslackworkflowapiDomain1F140300",
        },
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
        "Stage": Object {
          "Ref": "infraslackworkflowapiDeploymentStageprodD67E1A6B",
        },
      },
      "Type": "AWS::ApiGateway::BasePathMapping",
    },
    "infraslackworkflowapieventhandler2B3DE92B": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "infraslackworkflowapiB910C56A",
            "RootResourceId",
          ],
        },
        "PathPart": "event-handler",
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "infraslackworkflowapieventhandlerPOSTA78C202D": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:us-east-1:lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "InfraSlackWorkflowLambda833DA3E4",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "infraslackworkflowapieventhandler2B3DE92B",
        },
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "infraslackworkflowapieventhandlerPOSTApiPermissionTestinfraslackworkflowinfraslackworkflowapi7A68E075POSTeventhandlerBE572958": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "InfraSlackWorkflowLambda833DA3E4",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-east-1:572724207364:",
              Object {
                "Ref": "infraslackworkflowapiB910C56A",
              },
              "/test-invoke-stage/POST/event-handler",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "infraslackworkflowapieventhandlerPOSTApiPermissioninfraslackworkflowinfraslackworkflowapi7A68E075POSTeventhandlerE37B4E61": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "InfraSlackWorkflowLambda833DA3E4",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-east-1:572724207364:",
              Object {
                "Ref": "infraslackworkflowapiB910C56A",
              },
              "/",
              Object {
                "Ref": "infraslackworkflowapiDeploymentStageprodD67E1A6B",
              },
              "/POST/event-handler",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
}
`;

exports[`application with version 1`] = `
Object {
  "Outputs": Object {
    "LoadBalancerDNS": Object {
      "Value": Object {
        "Ref": "infraslackworkflowapiDomainDnsRecordCFDC8FA1",
      },
    },
    "infraslackworkflowapiEndpoint05FBE7B7": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "infraslackworkflowapiB910C56A",
            },
            ".execute-api.us-east-1.",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "infraslackworkflowapiDeploymentStageprodD67E1A6B",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecArtifactHashF1D0A81E": Object {
      "Description": "Artifact hash for asset \\"933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ec\\"",
      "Type": "String",
    },
    "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3BucketB74DFC99": Object {
      "Description": "S3 bucket for asset \\"933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ec\\"",
      "Type": "String",
    },
    "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3VersionKeyCAADF052": Object {
      "Description": "S3 key for asset version \\"933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ec\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "InfraSlackWorkflowLambda833DA3E4": Object {
      "DependsOn": Array [
        "InfraSlackWorkflowLambdaServiceRoleDefaultPolicy94F58139",
        "InfraSlackWorkflowLambdaServiceRole75DBBB72",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3BucketB74DFC99",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3VersionKeyCAADF052",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters933c589f693365f6831c085f7c5bc7cff1eb0b04c5260dbb24c7bd48f9fb15ecS3VersionKeyCAADF052",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {},
        },
        "Handler": "index.handler",
        "MemorySize": 3008,
        "Role": Object {
          "Fn::GetAtt": Array [
            "InfraSlackWorkflowLambdaServiceRole75DBBB72",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "InfraSlackWorkflowLambdaEventInvokeConfigD2210958": Object {
      "Properties": Object {
        "FunctionName": Object {
          "Ref": "InfraSlackWorkflowLambda833DA3E4",
        },
        "MaximumRetryAttempts": 0,
        "Qualifier": "$LATEST",
      },
      "Type": "AWS::Lambda::EventInvokeConfig",
    },
    "InfraSlackWorkflowLambdaServiceRole75DBBB72": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "InfraSlackWorkflowLambdaServiceRoleDefaultPolicy94F58139": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ssm:GetParameter",
                "ssm:GetParameters",
              ],
              "Effect": "Allow",
              "Resource": Array [
                "arn:aws:ssm:us-east-1:572724207364:parameterundefined",
              ],
            },
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::ImportValue": "infra-slack-workflow-dynamodb-version:ExportsOutputFnGetAttinfraslackworkflowdynamodbCB872367Arn36B4C1E3",
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "InfraSlackWorkflowLambdaServiceRoleDefaultPolicy94F58139",
        "Roles": Array [
          Object {
            "Ref": "InfraSlackWorkflowLambdaServiceRole75DBBB72",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "infraslackworkflowapiB910C56A": Object {
      "Properties": Object {
        "Description": "API for the InfraWorkflow App",
        "EndpointConfiguration": Object {
          "Types": Array [
            "REGIONAL",
          ],
        },
        "Name": "infra-slack-workflow-api",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "infraslackworkflowapiDeployment97D3C2AFa0112814ea588aefd026eba378ffe5f2": Object {
      "DependsOn": Array [
        "infraslackworkflowapieventhandlerPOSTA78C202D",
        "infraslackworkflowapieventhandler2B3DE92B",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "infraslackworkflowapiDeploymentStageprodD67E1A6B": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "infraslackworkflowapiDeployment97D3C2AFa0112814ea588aefd026eba378ffe5f2",
        },
        "MethodSettings": Array [
          Object {
            "DataTraceEnabled": false,
            "HttpMethod": "*",
            "LoggingLevel": "INFO",
            "MetricsEnabled": true,
            "ResourcePath": "/*",
          },
        ],
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
        "StageName": "prod",
        "TracingEnabled": true,
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "infraslackworkflowapiDomain1F140300": Object {
      "Properties": Object {
        "DomainName": "infraslackworkflowstack.cvent-sandbox.cvent.cloud",
        "EndpointConfiguration": Object {
          "Types": Array [
            "REGIONAL",
          ],
        },
        "RegionalCertificateArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":acm:us-east-1:572724207364:certificate/56e71c9e-f5a0-47d7-9c10-a11c0fbbf502",
            ],
          ],
        },
      },
      "Type": "AWS::ApiGateway::DomainName",
    },
    "infraslackworkflowapiDomainDnsRecordCFDC8FA1": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "infraslackworkflowapiDomain1F140300",
              "RegionalDomainName",
            ],
          },
          "HostedZoneId": Object {
            "Fn::GetAtt": Array [
              "infraslackworkflowapiDomain1F140300",
              "RegionalHostedZoneId",
            ],
          },
        },
        "HostedZoneId": "Z321UXK1TZE4RD",
        "Name": "infraslackworkflowstack.cvent-sandbox.cvent.cloud.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "infraslackworkflowapiDomainMapinfraslackworkflowversioninfraslackworkflowapi95C389DA6FBDE55F": Object {
      "Properties": Object {
        "DomainName": Object {
          "Ref": "infraslackworkflowapiDomain1F140300",
        },
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
        "Stage": Object {
          "Ref": "infraslackworkflowapiDeploymentStageprodD67E1A6B",
        },
      },
      "Type": "AWS::ApiGateway::BasePathMapping",
    },
    "infraslackworkflowapieventhandler2B3DE92B": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "infraslackworkflowapiB910C56A",
            "RootResourceId",
          ],
        },
        "PathPart": "event-handler",
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "infraslackworkflowapieventhandlerPOSTA78C202D": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:us-east-1:lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "InfraSlackWorkflowLambda833DA3E4",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "infraslackworkflowapieventhandler2B3DE92B",
        },
        "RestApiId": Object {
          "Ref": "infraslackworkflowapiB910C56A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "infraslackworkflowapieventhandlerPOSTApiPermissionTestinfraslackworkflowversioninfraslackworkflowapi95C389DAPOSTeventhandlerA08A17D3": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "InfraSlackWorkflowLambda833DA3E4",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-east-1:572724207364:",
              Object {
                "Ref": "infraslackworkflowapiB910C56A",
              },
              "/test-invoke-stage/POST/event-handler",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "infraslackworkflowapieventhandlerPOSTApiPermissioninfraslackworkflowversioninfraslackworkflowapi95C389DAPOSTeventhandlerEB0D153C": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "InfraSlackWorkflowLambda833DA3E4",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-east-1:572724207364:",
              Object {
                "Ref": "infraslackworkflowapiB910C56A",
              },
              "/",
              Object {
                "Ref": "infraslackworkflowapiDeploymentStageprodD67E1A6B",
              },
              "/POST/event-handler",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
}
`;
